### PR TITLE
[LOOP-1657] insulin model selection unit display updates as expected

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/LOOP-1657

Note that this fix will be landing in `release/v1.1.0`

### CODE REVIEW

File: LoopKitUI/Views/Settings Editors/InsulinModelSelection.swift [hunk 1]
```diff
@@ -11,18 +11,19 @@ import SwiftUI
...
-    let glucoseUnit: HKUnit
-    let supportedModelSettings: SupportedInsulinModelSettings
-    let mode: SettingsPresentationMode
-    let save: (_ insulinModelSettings: InsulinModelSettings) -> Void
-    let chartManager: ChartsManager
+    @Environment(\.dismissAction) private var dismiss
+    @Environment(\.authenticate) private var authenticate
+
+    @State private var value: InsulinModelSettings
+    @State private var chartManager: ChartsManager
```
Comment (darinkrauss) [inline LoopKitUI/Views/Settings Editors/InsulinModelSelection.swift:20]: [nit] Was making `chartManager` a `@State` intentional?

File: LoopKitUI/Views/Settings Editors/InsulinModelSelection.swift [hunk 2]
```diff
@@ -11,18 +11,19 @@ import SwiftUI
...
+
+    private let initialValue: InsulinModelSettings
+    private let insulinSensitivitySchedule: InsulinSensitivitySchedule
+    private let supportedInsulinModelSettings: SupportedInsulinModelSettings
+    private let mode: SettingsPresentationMode
+    private let saveInsulinModelSelection: (_ insulinModelSettings: InsulinModelSettings) -> Void
```
Comment (darinkrauss) [inline LoopKitUI/Views/Settings Editors/InsulinModelSelection.swift]: [nit] Why the change from `save` to `saveInsulinModelSelection`?

**Review by darinkrauss — APPROVED** [review]
A couple of nits, but otherwise LGTM!

File: LoopKitUI/Views/Settings Editors/InsulinModelSelection.swift (hunk 1 above)
Comment (nhamming) [reply LoopKitUI/Views/Settings Editors/InsulinModelSelection.swift:20]: Yes, since a new chartManager was being creating with each view update, the reference to one controlling the displayed chart unit was lost and thus it could not be used to triggered a chart update.

File: LoopKitUI/Views/Settings Editors/InsulinModelSelection.swift (hunk 2 above)
Comment (nhamming) [reply LoopKitUI/Views/Settings Editors/InsulinModelSelection.swift]: No real reason, I just find it clearer. I'll revert.
